### PR TITLE
Explicit lock

### PR DESCRIPTION
**Pull request by remyrd**
Added:
- Keybinding to lock explicitly
- Option to disable auto-lock

Master Password too long + no one knows how I launch bwmenu ¯\_(ツ)_/¯

### CODE REVIEW

File: bwmenu [hunk 1]
```diff
@@ -149,6 +152,7 @@ on_rofi_exit() {
     12) show_urls;;
     13) show_folders;;
     17) copy_totp "$2";;
+    18) keyctl purge user bw_session &>/dev/null;;
```
Comment (Mange) [inline bwmenu]: I think a named function would fit better here considering all other actions are named functions.

**Review by Mange — COMMENTED** [review]
I'd be fine with this. How about you @mattydebie?

**Comment by mattydebie** [comment]
Actually I'm a little hesitant.
If we check the man pages for *keyctl*:
```bash
keyctl timeout <key> <timeout>

       This command is used to set the timeout on a key, or clear an existing timeout if the value specified is zero. The timeout is given as a number of seconds into the future.

              $ keyctl timeout $1 45
```

I'm thinking we should change the autolock to be: 
  * _-1_: lock immediatly after use
  * _0_: never lock (this will be passed straight to keyctl, default functionality)
  * _>0_: seconds to timeout after

I should've checked this while implementing keyctl, as this would have been added already..
What do you guys think? @Mange @remyrd

**Comment by remyrd** [comment]
> I'm thinking we should change the autolock to be:
> 
>     * _-1_: lock immediatly after use
>     * _0_: never lock (this will be passed straight to keyctl, default functionality)
>     * _>0_: seconds to timeout after

I thought about it but decided to keep it this way for 2 reasons.

1. It makes more sense to me (semantically) that locking immediately is equivalent to 0 seconds, thus -1 as nonsensical == disable
2. Users randomly updating would suddenly have a perma-open vault, believing it is locked. Which would, after making them confused (and maybe angry), force them to rewrite the command/macro/hotkey they launch the script with.

In any case, another option is to add something like `--ignore-auto-lock`.
Bottom line I intended to add a feature without disturbing anyone. 
Thoughts? @Mange @mattydebie

**Comment by mattydebie** [comment]
> Users randomly updating would suddenly have a perma-open vault, believing it is locked

Yeah I thought about this too. Although this is still a 0.x release, this might disturb some people. 
If you squash your commits, adjust the part @Mange commented about and add your name to the README's contributers (alphabetically please), I'll merge this PR :grin:

**Comment by Mange** [comment]
I also think `0` makes more sense for "0 seconds" and `-1` meaning "just ignore this bad value and disable it". If we want to be crystal clear, we could disallow value smaller than 0 and have an `--keep-open` option, but I think this is less complicated and not too hard for anyone to understand.
